src/GovUk.Education.ExploreEducationStatistics.Admin/Services/DataBlockService.cs

```diff
@@ -122,9 +122,34 @@ public class DataBlockService : IDataBlockService
                 {
                     var viewModel = _mapper.Map<DataBlockViewModel>(dataBlock);
 
+                    var subjectId = dataBlock.Query.SubjectId;
+
+                    // TODO luke: consolidate into one query instead of 3 seperate calls
+                    // to the database
+
+                    var releaseId = await _context.ReleaseContentBlocks
+                        .Where(rcb => rcb.ContentBlockId == id)
+                        .Select(rcb => rcb.ReleaseId)
+                        .SingleAsync();
```

Comment:
You can get the `releaseId` without a query since we've already called `GetReleaseContentBlock`. 

To get it in this `OnSuccess` block, you can change `OnSuccess(CheckIsDataBlock)` above to `OnSuccessDo(CheckIsDataBlock)`. This means that the `releaseContentBlock` returned from `GetReleaseContentBlock` is passed forward rather than the result of `CheckIsDataBlock`, meaning you can just do `var releaseId = releaseContentBlock.ReleaseId;`